src/geocat/comp/__init__.py

```
@@ -1105,4 +1105,248 @@ def rcm2points(lat2d, lon2d, fi, lat1dPoints, lon1dPoints, opt=0, msg=None, meta
+
+	msg (:obj:`numpy.number`):
+	    A numpy scalar value that represent a missing value in `z`.
+	    This argument allows a user to use a missing value scheme
+	    other than NaN or masked arrays, similar to what NCL allows.
+
+	meta (:obj:`bool`):
+	    Set to False to disable metadata; default is False.
+
+    Returns:
+	:class:`numpy.ndarray`: If any argument is "double" the return type
+        will be "double"; otherwise a "float" is returned.
+
+    Description:
+        The maximum size of the returned array will be 3 x ld where ld <= ny*mx.
+        If no missing values are encountered in z, then ld=ny*mx. If missing
+        values are encountered in z, they are not returned and hence ld will be
+        equal to ny*mx minus the number of missing values found in z. The return
+        array will be double if any of the input arrays are double, and float
+        otherwise.
+
+    Examples:
+
+	Example 1: Using grid2triple with :class:`xarray.DataArray` input
+
+	.. code-block:: python
+
+	    import numpy as np
+	    import xarray as xr
+	    import geocat.comp
+
+	    # Open a netCDF data file using xarray default engine and load the data stream
+	    ds = xr.open_dataset("./NETCDF_FILE.nc")
+
+	    # [INPUT] Grid & data info on the source curvilinear
+	    z=ds.DIST_236_CBL[:]
+	    x=ds.gridlat_236[:]
+	    y=ds.gridlon_236[:]
+
+	    output = geocat.comp.grid2triple(x, y, z)
+    """
+
+    # Basic sanity checks
+    if z.ndim != 2 :
+        raise DimensionError("ERROR grid2triple: `z` must be two dimensions !\n")
+
+    if isinstance(x, xr.DataArray):
+        x = x.values
+    if isinstance(y, xr.DataArray):
+        y = y.values
+    if isinstance(z, xr.DataArray):
+        z = z.values
+
+    if isinstance(z, np.ndarray):
+        fo = _ncomp._grid2triple(x, y, z, msg)
+    else:
+        raise TypeError
+
+    if meta and isinstance(input, xr.DataArray):
+        pass	 # TODO: Retaining possible metadata might be revised in the future
```

Same comment here wrt silently ignoring this option.

---

https://github.com/NCAR/geocat-comp/pull/53/commits/0d8d4f53013fd903ee03df52ace649a58ea354fa